prov/sockets/src/sock_progress.c

```diff
@@ -2520,10 +2517,11 @@ static void sock_pe_wait(struct sock_pe *pe)
 {
 	char tmp;
 	int ret;
+	void *ep_contexts[SOCK_EPOLL_WAIT_EVENTS];
 
-	ret = sock_epoll_wait(&pe->epoll_set, -1);
-        if (ret < 0)
-                SOCK_LOG_ERROR("poll failed : %s\n", strerror(errno));
+	ret = fi_epoll_wait(pe->epoll_set, ep_contexts, SOCK_EPOLL_WAIT_EVENTS, -1);
```

Comment:
Please see my previous comment:
> I think that one of the reason to set the maximum events to more than 1 is for efficiently: you can return multiple events with one single syscall and amortize the overhead induced by repetitive calls to epoll_wait().

I think that returning multiple events with one call to epoll_wait() is more efficient that returning the events one by one (e.g., repetitive calls to epoll_wait() with 1 single event).